src/openapi/transport/retry.js

```diff
@@ -130,29 +139,63 @@ TransportRetry.prototype.sendTransportCall = function(transportCall) {
  * @param transportCall
  */
 TransportRetry.prototype.addFailedCall = function(transportCall) {
-    transportCall.retryCount++;
     this.failedCalls.push(transportCall);
-    if (!this.retryTimer) {
+    const callOptions = this.methods[transportCall.method];
+    if (callOptions.retryTimeouts && callOptions.retryTimeouts.length > transportCall.retryCount) {
+        // schedule an individual retry timeout
+        transportCall.retryTimer = setTimeout(() => {
+            this.retryIndividualFailedCall(transportCall);
+        }, callOptions.retryTimeouts[transportCall.retryCount]);
+    } else if (!this.retryTimer) {
+        // schedule a retry timeout for all failed calls
         this.retryTimer = setTimeout(() => {
             this.retryFailedCalls();
         }, this.retryTimeout);
     }
+    transportCall.retryCount++;
 };
 
 /**
  * @protected
  */
 TransportRetry.prototype.retryFailedCalls = function() {
     this.retryTimer = null;
-    while (this.failedCalls.length > 0) {
-        this.sendTransportCall(this.failedCalls.shift());
+    let i = 0;
+    while (i < this.failedCalls.length) {
+        const currentFailedCall = this.failedCalls[i];
+        if (currentFailedCall.retryTimer) {
+            // skip individually scheduled calls
+            i++;
+        } else {
+            this.failedCalls.splice(i, 1);
+            this.sendTransportCall(currentFailedCall);
+        }
     }
 };
 
 /**
- * Disposes the underlying transport, the failed calls queue and clears retry timer.
+ * @protected
+ * @param transportCall
+ */
+TransportRetry.prototype.retryIndividualFailedCall = function(transportCall) {
+    transportCall.retryTimer = null;
+    const failedCallsIndex = this.failedCalls.indexOf(transportCall);
+    if (failedCallsIndex >= 0) {
+        this.failedCalls.splice(failedCallsIndex, 1);
+    }
+    this.sendTransportCall(transportCall);
+};
+
+/**
+ * Disposes the underlying transport, the failed calls queue and clears retry timers.
  */
 TransportRetry.prototype.dispose = function() {
+    this.failedCalls.forEach((failedCall) => {
+        if (failedCall.retryTimer) {
+            failedCall.retryTimer.clear();

```

Comment:
ah it needs to be clearTimeout(), there's a little bug below this then where it does this.retryTimer.clear()